packages/sdk-core/src/modules/metrics/MetricsBuilder.ts

```diff
@@ -0,0 +1,137 @@
+import { TimeHelper } from '../../common/TimeHelper';
+import { BacktraceConfiguration, BacktraceMetricsOptions } from '../../model/configuration/BacktraceConfiguration';
+import { BacktraceRequestHandler } from '../../model/http';
+import { AttributeManager } from '../attribute/AttributeManager';
+import { BacktraceMetrics } from './BacktraceMetrics';
+import { BacktraceSessionProvider } from './BacktraceSessionProvider';
+import { MetricsQueue } from './MetricsQueue';
+import { MetricsSubmissionQueue } from './MetricsSubmissionQueue';
+import { MetricsUrlInformation } from './MetricsUrlInformation';
+import { SummedEvent } from './model/SummedEvent';
+import { UniqueEvent } from './model/UniqueEvent';
+
+interface ApplicationInfo {
+    application: string;
+    applicationVersion: string;
+}
+export class MetricsBuilder {
+    /**
+     * Default metrics submission interval. The variable defines how often metrics will be sent to metrics system
+     * By default 30 mins.
+     */
+    public readonly DEFAULT_UPDATE_INTERVAL = TimeHelper.convertSecondsToMilliseconds(30 * 60);
+
+    private readonly APPLICATION_VERSION_ATTRIBUTE = 'application.version';
+    private readonly APPLICATION_ATTRIBUTE = 'application';
+    constructor(
+        private readonly _options: BacktraceConfiguration,
+        private readonly _sessionProvider: BacktraceSessionProvider,
+        private readonly _attributeManager: AttributeManager,
+        private readonly _requestHandler: BacktraceRequestHandler,
+    ) {}
+
+    public build(
+        uniqueEventsSubmissionQueue?: MetricsQueue<UniqueEvent>,
+        summedEventsSubmissionQueue?: MetricsQueue<UniqueEvent>,
+    ): BacktraceMetrics | undefined {
+        const metricsOptions = {
+            ...this.optionsWithDefaults(),
+            ...(this._options.metrics ?? {}),
+        };
+        if (!metricsOptions.enable) {
+            return undefined;
+        }
+        const applicationInfo = this.verifyAttributeSetup();
+        if (!applicationInfo) {
+            return undefined;
+        }
+
+        uniqueEventsSubmissionQueue =
+            uniqueEventsSubmissionQueue ??
+            this.createUniqueEventSubmissionQueue(metricsOptions.metricsSubmissionUrl as string, applicationInfo);
+        if (!uniqueEventsSubmissionQueue) {
+            return undefined;
+        }
+
+        summedEventsSubmissionQueue =
+            summedEventsSubmissionQueue ??
+            this.createSummedEventSubmissionQueue(metricsOptions.metricsSubmissionUrl as string, applicationInfo);
+        if (!summedEventsSubmissionQueue) {
+            return undefined;
+        }
```

Comment:
Maybe it's worth to give back some information about why metrics were not created?

Reply:
It makes a lot of sense but the SDK right now doesn't have any logger or any logger options. Can we handle it in the future when we will be working on options related to logging?

Reply:
sure